### PR TITLE
Always use gzip for squashfs compression

### LINKED ISSUE #2717: RFE: Always use gzip for Squashfs
## Version of Singularity:

For 3.2.0

### Expected behavior

Always use gzip.

### Actual behavior

Mksquashfs defaults to host settings.

### PR DESCRIPTION
**Pull request by ikaneshiro**
**This fixes or addresses the following GitHub issues:**

- Fixes #2717 

**Before submitting a PR, make sure you have done the following:**

- Read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/master/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)

Attn: @singularity-maintainers

### CODE REVIEW

**Review by WestleyK — COMMENTED** [review]
Line: `177`, `172`, `168`, `163`, `159`, `139`, and `118` all have caps in the return error message. I know its outside your changes, but it should probably be fixed. Otherwise, looks good :+1:

**Comment by ikaneshiro** [comment]
Marking WIP for now since versions of `mksquashfs` less than 4.1 do not have the flag I was using. Will need to add a check.

**Review by mem — COMMENTED** [review]
LGTM, I haven't approved because it still has the WIP flag, so I don't know if there's pending work?

**Comment by jmstover** [comment]
This will not work ... See `mksquashfs` on SLES 11
```
# mksquashfs --help 2>&1 | grep 'comp'
-noI                    do not compress inode table
-noD                    do not compress data blocks
-noF                    do not compress fragment blocks
```

There are versions of mksquashfs that do not have the `-comp` flag.

Generally, I believe these versions *only* support gzip though. At least I've never seen one where it's supported something else.

File: internal/pkg/build/assemblers/assembler_sif.go [hunk 1]
```diff
@@ -146,7 +146,7 @@ func (a *SIFAssembler) Assemble(b *types.Bundle, path string) (err error) {
 	os.Remove(squashfsPath)
 	defer os.Remove(squashfsPath)
 
-	args := []string{b.Rootfs(), squashfsPath, "-noappend"}
+	args := []string{b.Rootfs(), squashfsPath, "-noappend", "-comp", "gzip"}
```
Comment (jmstover) [inline internal/pkg/build/assemblers/assembler_sif.go]: There are versions of mksquashfs on distro's we support that do not have the `-comp` flag

https://github.com/sylabs/singularity/pull/3215#issuecomment-479684113

**Review by jmstover — CHANGES_REQUESTED** [review]

**Pull request closed without merging**